### PR TITLE
[Fixes #3503] Removing formatters on a given type only works on collections in MVC options

### LINKED ISSUE #3503: Removing formatters on a given type only works on collections in MVC options
Currently a formatter can be removed by using the [`RemoveType`](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/Formatters/FormatterCollection.cs#L19) method on the formatters collection on MVC options 

Example: `options.OutputFormatters.RemoveType<StringOutputFormatter>()`

We have collection of formatters exposed through other areas of the framework like [`ResourceExecutingContext`](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Abstractions/Filters/ResourceExecutingContext.cs#L35) where this method cannot be used. An example scenario here is a user trying to have a per-controller specific list of formatters, which he can achieve using  a resource filter

Should we have the formatter collections on MVC options to be just `List<T>` and have an extension method on `IList<T>` which can achieve this scenario? I know this is very generic and need not apply to formatters only

Example:
```
public static class MyExtensions
{
    public static void RemoveType<T>(this IList<T> list, Type itemType)
    {
        var itemTypeInfo = itemType.GetTypeInfo();
        for (var i = list.Count - 1; i >= 0; i--)
        {
            var item = list[i];
            if (itemTypeInfo.IsAssignableFrom(item.GetType().GetTypeInfo()))
            {
                list.RemoveAt(i);
            }
        }
    }
}
```

One observation...`FormatterCollection` is under `Mvc.Formatters` namespace which makes discovering `RemoveType<>` hard

### PR DESCRIPTION
**Pull request by kichalla**
@rynowak

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Abstractions/Formatters/FormatterCollection.cs [hunk 1]
```diff
@@ -11,6 +12,23 @@ namespace Microsoft.AspNet.Mvc.Formatters
     /// <typeparam name="TFormatter">The type of formatters in the collection.</typeparam>
     public class FormatterCollection<TFormatter> : Collection<TFormatter>
```
Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.Abstractions/Formatters/FormatterCollection.cs]: Just FYI...I moved this type from Mvc.Core to Mvc.Abstractions

File: test/WebSites/FiltersWebSite/Controllers/SpecificFormattersController.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.Filters;
+using Microsoft.AspNet.Mvc.Formatters;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace FiltersWebSite
```
Comment (rynowak) [inline test/WebSites/FiltersWebSite/Controllers/SpecificFormattersController.cs]: Can you put this in `BasicWebSite` instead? We might mothball this site. This is a reasonable test, and I don't want it to get lost in the scrum.

File: test/WebSites/FiltersWebSite/Controllers/SpecificFormattersController.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            if (!ModelState.IsValid)
+            {
+                return HttpBadRequest(ModelState);
+            }
+
+            return Ok(person);
+        }
+
+        [HttpPost("noncamelcase")]
+        public IActionResult Post2([FromBody] Person person)
```
Comment (rynowak) [inline test/WebSites/FiltersWebSite/Controllers/SpecificFormattersController.cs]: Better names for these?

File: src/Microsoft.AspNet.Mvc.Abstractions/Formatters/FormatterCollection.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Abstractions/Formatters/FormatterCollection.cs]: Seems alright.

**Comment by rynowak** [comment]
:watch:

File: test/WebSites/FiltersWebSite/Controllers/SpecificFormattersController.cs (hunk 2 above)
Comment (kichalla) [inline test/WebSites/FiltersWebSite/Controllers/SpecificFormattersController.cs]: Sure will do

**Comment by kichalla** [comment]
:up: :date:

**Comment by rynowak** [comment]
:shipit:

**Comment by kichalla** [comment]
982c7abea8fc540142dc6566353508f42c4dc352

**Pull request closed without merging**